src/core/model.js

```diff
@@ -262,6 +280,38 @@ export default class Model extends Object3D {
       instanceCount
     });
 
+    if (this.timerQueryEnabled === true) {
+      if (this.lastQueryReturned === true) {
+        this.ext.endQueryEXT(this.ext.TIME_ELAPSED_EXT);
+        this.profileFrameCount++;
+        this.lastQueryReturned = false;
+      }
+  // ...at some point in the future, after returning control to the browser and being called again:
+      const disjoint = this.program.gl.getParameter(this.ext.GPU_DISJOINT_EXT);
+      if (disjoint) {
+        this.lastQueryReturned = true;
+        // Have to redo all of the measurements.
+      } else {
+        const available = this.ext.getQueryObjectEXT(this.timeElapsedQuery,
+          this.ext.QUERY_RESULT_AVAILABLE_EXT);
+
+        if (available) {
+          const timeElapsed = this.ext.getQueryObjectEXT(this.timeElapsedQuery,
+            this.ext.QUERY_RESULT_EXT) / 1e6;
+          this.accumulatedFrameTime += timeElapsed;
+          this.averageFrameTime = this.accumulatedFrameTime / this.profileFrameCount;
+          // Do something useful with the time.  Note that care should be
+          // taken to use all significant bits of the result, not just the
+          // least significant 32 bits.
+          log.log(2, 'program.id: ', this.program.id);
```

Comment:
+1.

For the long-term, a nice way to handle this would be to update a global `perf` or `stats` object, and make a stats widget that can show this in the UI.

There is three.js stats widget and regl-stats widget etc we could investigate, maybe we could use directly.